modules/queue/queue_channel.go

```diff
@@ -126,6 +126,10 @@ func (q *ChannelQueue) FlushWithContext(ctx context.Context) error {
 				log.Error("Unhandled Data whilst flushing queue %d", q.qid)
 			}
 			atomic.AddInt64(&q.numInQueue, -1)
+		default:
+			continue
+		}
+		select {
```

Comment:
This change will prevent flushing from being cancelled.

We have to be able to cancel the flush.

Reply:
How does flush cancellation work, which channels are active?

First select block checks two channels and move through default statement, if none are active. Then second select being checked.
This would check all channels and give priority to first two.

Reply:
Oh, i got it, there should be no continue.